components/org.wso2.carbon.identity.conditional.auth.typingdna.api/src/main/java/org/wso2/carbon/identity/conditional/auth/typingdna/api/core/TypingDNAServiceImpl.java

```
@@ -0,0 +1,143 @@
+        String tenantDomain = "carbon.super";
+        if (IdentityUtil.threadLocalProperties.get().get("TenantNameFromContext") != null) {
+            tenantDomain = (String) IdentityUtil.threadLocalProperties.get().get("TenantNameFromContext");
+        }
+        String userID = getUserID(username, tenantDomain);
+
+        try {
+
+            String APIKey = CommonUtils.getConnectorConfig("adaptive_authentication.tdna.username", tenantDomain);
+            String APISecret = CommonUtils.getConnectorConfig("__secret__adaptive_authentication.tdna.password", tenantDomain);
+            String region = CommonUtils.getConnectorConfig("adaptive_authentication.tdna.apiregion", tenantDomain);
+
+            String baseurl = buildURL(region, userID);
+            String Authorization = Base64.getEncoder().encodeToString((APIKey + ":" + APISecret).getBytes(StandardCharsets.UTF_8));
+            URL url = new URL(baseurl);
+            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+            connection.setRequestMethod("DELETE");
+            connection.setConnectTimeout(5000);
+            connection.setReadTimeout(5000);
+            connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+            connection.setRequestProperty("Authorization", "Basic " + Authorization);
+
+            connection.setUseCaches(false);
+            connection.setDoOutput(true);
+
+            InputStream is = connection.getInputStream();
+            BufferedReader rd = new BufferedReader(new InputStreamReader(is));
+            StringBuilder res = new StringBuilder();
+            String line;
+            while ((line = rd.readLine()) != null) {
+                res.append(line);
+                res.append('\r');
+            }
+            rd.close();
+        } catch (MalformedInputException e) {
+            log.error(e);
+        } catch (IdentityEventException e) {
+            log.error(e);
```

Need to throw a proper exception from here and handle it in the places that use this method.